ansible-ipi-install/roles/installer/tasks/24_rhcos_image_cache.yml

```diff
@@ -0,0 +1,124 @@
+---
+# rhcos_json fact already set in 23_rhcos_image_paths.yaml
+
+- name: Set fact for RHCOS_QEMU_SHA256 (4.4+ only)
+  set_fact:
+    rhcos_qemu_sha256: '{{ rhcos_json.json | json_query(''images.qemu."uncompressed-sha256"'') }}'
+  when: (release_version[0]|int > 4) or (release_version[2]|int > 3)
+
+- name: Set fact for RHCOS_OPENSTACK_SHA256
+  set_fact:
+    rhcos_openstack_sha256: "{{ rhcos_json.json | json_query('images.openstack.sha256') }}"
+
+- name: Check if {{ rhcos_qemu_uri }} is already in cache (4.4+ only)
+  stat: 
+    path: "/var/www/html/{{ rhcos_qemu_uri }}"
+    checksum_algorithm: sha256
+    get_checksum: yes
+  register: rhcos_qemu_stat
+  when: (release_version[0]|int > 4) or (release_version[2]|int > 3)
+
+- name: Download {{ rhcos_qemu_uri }} for cache (4.4+ only)
+  get_url:
+    url:  "{{ rhcos_path }}/{{ rhcos_qemu_uri }}"
+    dest: "/var/www/html/{{ rhcos_qemu_uri }}"
+    owner: "{{ ansible_user }}"
+    group: "{{ ansible_user }}"
+    mode: '0755'
+  when: ((release_version[0]|int > 4) or (release_version[2]|int > 3)) and (rhcos_qemu_stat.stat.exists == false or rhcos_qemu_stat.stat.checksum != rhcos_qemu_sha256)
+  become: yes
```

Comment:
because you use `ansible_user` here and become: yes, this will make the owner and group root/root (FYI). Is this OK?

Reply:
It does do as you note, but it doesn't cause a problem.  I wanted to save a couple plays for setting the permissions, but we can add them if you think it would be wise.